websocket.el

```diff
@@ -731,17 +731,19 @@ to the websocket protocol.
         (websocket-debug websocket "State change to %s" change)
         (let ((status (process-status process)))
           (when (and nowait (eq status 'open))
-            (websocket-ensure-handshake url conn key protocols extensions custom-header-alist))
+            (websocket-ensure-handshake url conn key protocols extensions custom-header-alist nowait))
 
           (when (and (member status '(closed failed exit signal))
                      (not (eq 'closed (websocket-ready-state websocket))))
             (websocket-try-callback 'websocket-on-close 'on-close websocket))))))
 
-(defun websocket-ensure-handshake (url conn key protocols extensions custom-header-alist)
+(defun websocket-ensure-handshake (url conn key protocols extensions custom-header-alist nowait)
   (let ((url-struct (url-generic-parse-url url))
         (websocket (process-get conn :websocket)))
-    (when (and (eq 'connecting (websocket-ready-state websocket))
-               (eq 'open (process-status conn)))
+    (when(and (eq 'connecting (websocket-ready-state websocket))
+              (memq (process-status conn)
+                    (or (and nowait '(connect run))
```

Comment:
I think this would be more straightforward to read if using an if instead of and or and and an.  Something like `(list 'run (if nowait 'connect 'open))` should be equivalent.